app/services/marketplace_service/api/marketplaces.rb

```diff
@@ -70,6 +70,62 @@ def create(params)
       return from_model(community)
     end
 
+    def set_locales(community, locales)
+      default_locale = community.locales[0]
+      community_name = community.name(default_locale)
+      locales.each { |locale| Helper.first_or_create_community_customization!(community, community_name, locale) }
+
+      settings = community.settings || {}
+      settings["locales"] = locales
+      community.settings = settings
+      community.save!
+
+      ensure_translations(community, locales).map { |translations|
+        locale = translations[:locale]
+        translations[:translations].each{ |translation|
+          add_translation(community, locale, translation[:translation_key], translation[:translation])
+        }
+      }
+      Rails.cache.delete("/translation_service/community/#{community.id}")
+    end
+
+    def ensure_translations(community, locales)
+      locales.map { |locale|
+        translations_res = TranslationService::API::Api.translations.get(community.id, {locales: [locale]})
```

Comment:
This way we could handle the creation of new dynamic translations for new languages a bit simpler.